tests/update/updates_packagekit_gpk.pm

```diff
@@ -14,13 +14,23 @@ use base "x11test";
 use strict;
 use testapi;
 use utils;
-use version_utils 'is_sle';
+use version_utils qw(is_sle is_leap);
 
 sub turn_off_screensaver {
+    my ($self) = @_;
     # Turn off screensaver
     x11_start_program('xterm');
     become_root;
     ensure_serialdev_permissions;
+
+    # Trying to detect bsc#1052258 which leads to the test failure occasionally on Leap 42.3
+    if (is_leap('=42.3') && script_run("dmesg | grep -P 'soft lockup - CPU#\\d+ stuck'")) {
+        # Fail on cpu stuck, see bsc#1052258 and collect logs if machine still responds
+        record_soft_failure 'cpu soft lockup detected, see bsc#1052258';
+        $self->save_upload_y2logs;
+        select_console 'x11';    # Trying to continue with the test
+    }
+
```

Comment:
can you comment why this code should be here where it is kinda not well fitting? I mean the method is called `turn_off_screensaver` so there should be at least a good name explaining why we have this here.

Reply:
and here I agree with you , probably we need to put it into some more reasonable place